omod/src/main/java/org/openmrs/module/fhir2/providers/EncounterFhirResourceProvider.java

```diff
@@ -69,7 +69,12 @@ public Bundle searchEncounter(@OptionalParam(name = Encounter.SP_DATE) DateRange
 	                Practitioner.SP_GIVEN, Practitioner.SP_FAMILY,
 	                Practitioner.SP_NAME }, targetTypes = Practitioner.class) ReferenceParam participantReference,
 	        @OptionalParam(name = Encounter.SP_SUBJECT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_GIVEN,
-	                Patient.SP_FAMILY, Patient.SP_NAME }, targetTypes = Patient.class) ReferenceParam subjectReference) {
+					Patient.SP_FAMILY, Patient.SP_NAME }, targetTypes = Patient.class) ReferenceParam subjectReference,
+			@OptionalParam(name = Encounter.SP_PATIENT, chainWhitelist = { "", Patient.SP_IDENTIFIER, Patient.SP_GIVEN,
+	                Patient.SP_FAMILY, Patient.SP_NAME }, targetTypes = Patient.class) ReferenceParam patientReference) {
+						if(patientReference != null){
+							subjectReference = patientReference;
+						}
```

Comment:
@reagan-meant I don't think that we should do this here. 

cc @ibacher

Reply:
Actually, this is explicitly what we do want to have happen. The `patient` search parameter is a sub-set of the `subject` search parameter where the `subject` is a Patient. So the `patient` parameter is more exact than the subject parameter. I suppose the other thing we could do is throw an exception here.

Reply:
@ibacher what would be the potential cause the exception?